2015/proposals/banaszkiewicz-piotr-amy.md

```diff
@@ -0,0 +1,214 @@
+# Enhance Amy, a workshop-management platform for Software Carpentry
+
+## Abstract
+
+The number of Software Carpentry's workshops run weekly dynamically grows. All
+the management for these workshops is currently being done by 1-2 persons and
+in very rough conditions (via files and text notes).  That's why Amy was
+created in December 2014: to help Software Carpentry admins schedule
+workshops, assign instructors to workshops, and even coordinate training for
+new instructors.
+
+The aim of this project is to continue enhancing Amy.  Many people
+[suggested great ideas](https://github.com/swcarpentry/amy/issues?q=is%3Aissue+is%3Aopen+label%3Aenhancement+-label%3Aessential) for Amy,
+some even [implemented them](https://github.com/swcarpentry/amy/pulls?q=is%3Apr+is%3Aopen+sort%3Acreated-asc), but not much was actually
+merged in -- and, as a result, in March
+[we can see a drop in new commits](https://github.com/swcarpentry/amy/graphs/contributors).
+
+## Features
+
+Integration that will allow Amy admins to easy control workshop-related
+services:
+
+* integration with Eventbrite
+    * checking if event's registration exists
+    * checking if event's registration is running
+    * maybe bulk-loading event attendees
+* integration with Mailman
+    * listing people signed up for a specific list
+    * listing lists person is signed up to
+    * creating lists for workshops for instructors and helpers and hosts
+* integration with GitHub
+    * OAuth login
+
+New functionality this project should bring into Amy:
+
+* reports (statistics)
+    1. number of people taught
+    2. number of instructors
+    3. number of workshops run to date
+    4. workshop enrolment per workshop
+    5. workshops without enrolment numbers
+    6. how many times an instructor has taught
+    7. list of instructors who have never taught
+    8. number of instructor trainees per cohort
+    9. new instructors from within N last months
```

Comment:
Maybe “from the last N months“?  I don't remember quite how badging works.  Do folks need to teach a workshop with a more experienced instructor before they get badged?  Or are they badged after completing the training course and landing a PR?  If it's the latter, we could make this “instructors who have been badged in the last N months”.